src/Npgsql/NpgsqlGlobalConfiguration.cs

```diff
@@ -0,0 +1,154 @@
+// DOCUMENTATION, EVEN IF THE NPGSQL DEVELOPMENT TEAM HAS BEEN ADVISED OF
+// THE POSSIBILITY OF SUCH DAMAGE.
+//
+// THE NPGSQL DEVELOPMENT TEAM SPECIFICALLY DISCLAIMS ANY WARRANTIES,
+// INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+// AND FITNESS FOR A PARTICULAR PURPOSE. THE SOFTWARE PROVIDED HEREUNDER IS
+// ON AN "AS IS" BASIS, AND THE NPGSQL DEVELOPMENT TEAM HAS NO OBLIGATIONS
+// TO PROVIDE MAINTENANCE, SUPPORT, UPDATES, ENHANCEMENTS, OR MODIFICATIONS.
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml;
+
+namespace Npgsql
+{
+    /// <summary>
+    /// Provides methods to access configuration data stored in a machine specific configuration file.
+    /// </summary>
+    internal class NpgsqlGlobalConfiguration
+    {
+        static NpgsqlGlobalConfiguration current;
+        /// <summary>
+        /// Returns a reference to the current NpgsqlGlobalConfiguration instance.
+        /// </summary>
+        public static NpgsqlGlobalConfiguration Current
+        {
+            get
+            {
+                if (current == null)
+                {
+                    current = new NpgsqlGlobalConfiguration();
+
+                    try
+                    {
+                        current.Load();
+                    }
+                    catch
+                    {
+                        System.Diagnostics.Trace.WriteLine("Failed to load global configuration.");
+                    }
+                }
+                return current;
+            }
+        }
+
+        IList<DatabaseInfoFactoryMetadata> databaseInfoFactories = new List<DatabaseInfoFactoryMetadata>();
+
+        /// <summary>
+        /// Returns the path of the configuration file.
+        /// </summary>
+        string ConfigFilePath => Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData), "Npgsql\\Npgsql_machine.config");
+        /// <summary>
+        /// Determines whether the configuration file exists.
+        /// </summary>
+        bool ConfigFileExists => File.Exists(ConfigFilePath);
+
+        /// <summary>
+        /// Returns an enumeration of DatabaseInfoFactoryMetadata instances which provide metadata for globally registered DatabaseInfoFactory implementations.
+        /// </summary>
+        public IEnumerable<DatabaseInfoFactoryMetadata> DatabaseInfoFactories
+        {
+            get
+            {
+                return databaseInfoFactories;
+            }
+        }
+        
+        void Load()
+        {
+            if (ConfigFileExists)
+            {
+                using (var reader = XmlReader.Create(ConfigFilePath))
+                {
+                    while(reader.Read())
+                    {
+                        if (reader.IsStartElement("databaseinfofactories"))
```

Comment:
Would it make sense to use a constant here other than pass it as string explicitly?

`const string MACHINE_CONFIG_FACTORY_TAG = "databaseinfofactories";`

Reply:
Nice. Feel free to merge.